modules/core/src/smithy4s/internals/package.scala

```diff
@@ -25,12 +25,16 @@ package object internals {
 
   private[internals] implicit class vectorOps[A](val vector: Vector[A])
       extends AnyVal {
-    def traverse[B](f: A => Option[B]): Option[Vector[B]] =
-      vector.foldLeft[Option[Vector[B]]](Some(Vector.empty)) { (result, a) =>
-        for {
-          acc <- result
-          b <- f(a)
-        } yield (acc.:+(b))
+    def traverse[B](f: A => Option[B]): Option[Vector[B]] = {
+      val vec = Vector.newBuilder[B]
+      var done = false
+      vector.takeWhile(_ => !done).foreach { a =>
```

Comment:
`takeWhile` creates an instance of `Vector` that will be instantly discarded. I'd prefer a normal `while` loop here

Reply:
Removed, but now I have an early return.. I've never used an early return in Scala before.. but it seems like it'd be the most efficient? Lmk if you want it done differently.

Reply:
Whoops, forgot local returns weren't allowed in scala 3.. updated to use a try catch.. idk like I said lmk if you want it done differently.

Reply:
Whoops, forgot local returns weren't allowed in scala 3.. updated to use a try catch.. idk like I said lmk if you want it done differently.

Reply:
even without stack the throw will have more overhead than an actual while loop.

Reply:
How do you feel about the latest version I posted? I can always do a while loop that I break out of instead.. or another trick instead of breaking..

Reply:
Agreeing to it because it's an official mechanism.